src/main/java/appeng/core/features/ItemFeatureHandler.java

```diff
@@ -105,15 +108,43 @@ else if( name.equals( "ItemPart" ) )
 
 			if( side == Side.CLIENT )
 			{
-				ModelBakery.registerItemVariants( item, registryName );
+				if( item instanceof AEBaseItem )
+				{
+					AEBaseItem baseItem = (AEBaseItem) item;
+
+					// Handle registration of item variants
+					ResourceLocation[] variants = baseItem.getItemVariants().toArray( new ResourceLocation[0] );
+					ModelBakery.registerItemVariants( item, variants );
+				}
+				else
+				{
+					ModelBakery.registerItemVariants( item, registryName );
+				}
 			}
 		}
 	}
 
 	@Override
 	public void registerModel()
 	{
-		Minecraft.getMinecraft().getRenderItem().getItemModelMesher().register( item, 0, new ModelResourceLocation( registryName, "inventory" ) );
+		ItemMeshDefinition meshDefinition = null;
+
+		// Register a custom item model handler if the item wants one
+		if( item instanceof AEBaseItem )
+		{
+			AEBaseItem baseItem = (AEBaseItem) item;
+			meshDefinition = baseItem.getItemMeshDefinition();
+		}
+
+		if( meshDefinition != null )
+		{
+			ItemModelMesher itemModelMesher = Minecraft.getMinecraft().getRenderItem().getItemModelMesher();
+			itemModelMesher.register( item, meshDefinition );
```

Comment:
I don't want to call getItemMeshDefinition() twice because someone might decide to do some heavy lifting in there (i.e. creating mappings or something like that). 
So if i don't split it, I'd have to repeat the normal registration path in the instanceof check.
Related question: Why isn't item restricted to AEBaseItem anyway?